src/test/assembly/62531-array-cmp.rs

```diff
@@ -0,0 +1,21 @@
+// Ensure the asm for array comparisons is properly optimized on x86.
+
+// min-llvm-version: 11
+// assembly-output: emit-asm
+// compile-flags: -C opt-level=2
+// only-x86_64
+
+#![crate_type = "lib"]
+
+// CHECK-LABEL: compare
+// CHECK: movb $1, %al
+// CHECK-NEXT: retq
+#[no_mangle]
+pub fn compare() -> bool {
+    let bytes = 12.5f32.to_ne_bytes();
+    bytes == if cfg!(target_endian = "big") {
```

Comment:
This is a `x86_64`-only test, so I don't think this branch is relevant?

Reply:
Ah, never mind, this is checking if we end up optimizing a unconditional branch, I guess.